packages/network/README.md

```diff
@@ -17,7 +17,7 @@
 ## API
 
 ```typescript
-export type Middleware = (request: Request, fetch: Fetch) => ReturnType<Fetch>;
+export type Middleware = (request: Request, next: (request: RequestInfo) => void) : Promise<void>;
```

Comment:
I think we need to roll these README changes back (as of @hjdivad's commit changing implementation)

Reply:
can you double check? The implementation has changed because the original doc didn't make sense as a middleware author.

Reply:
Ya, see https://github.com/data-eden/data-eden/pull/12/files#diff-b58486ba03120faa0e3e740dcd2075eda09212a78d8c408b61f9dc6734007927R32-R39 for a test example:

```ts
  const csrfMiddleware: Middleware = async (
    request: Request,
    next: NormalizedFetch
  ) => {
    request.headers.set('X-CSRF', 'a totally legit request');

    return next(request);
  };
```

The idea here is that each middleware returns `Promise<Response>`, which allows a middleware to access the response object. Returning `Promise<void>` means that a middleware could only munge/mutate things on `Request` right?

Reply:
updated readme with updated types.